From 0b48a1ab0e1aaf6a67c066ef4c475d5521e23553 Mon Sep 17 00:00:00 2001
From: adrianknorr <adrian.f.knorr@gmail.com>
Date: Sat, 5 Dec 2020 18:03:47 +0100
Subject: [PATCH] Add support for search bar to search tags without "tag:"

It was not possible to search for a tag without the prefix "tag:".
For a better usage, it would be nice to search tags without the
prefix.
With the prefix, it only searches the tags, like before.
---
 internal/database/sqlite.go | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/internal/database/sqlite.go b/internal/database/sqlite.go
index ef561d5..8f8f65d 100644
--- a/internal/database/sqlite.go
+++ b/internal/database/sqlite.go
@@ -240,12 +240,20 @@ func (db *SQLiteDatabase) GetBookmarks(opts GetBookmarksOptions) ([]model.Bookma
 		query += ` AND (b.url LIKE ? OR b.excerpt LIKE ? OR b.id IN (
 			SELECT docid id 
 			FROM bookmark_content 
-			WHERE title MATCH ? OR content MATCH ?))`
+			WHERE title MATCH ? OR content MATCH ?)) OR
+			b.id IN (
+			SELECT bt.bookmark_id
+			FROM bookmark_tag bt
+			LEFT JOIN tag t ON bt.tag_id = t.id
+			WHERE t.name IN(?)
+			GROUP BY bt.bookmark_id
+			HAVING COUNT(bt.bookmark_id) = 1)`
 
 		args = append(args,
 			"%"+opts.Keyword+"%",
 			"%"+opts.Keyword+"%",
 			opts.Keyword,
+			opts.Keyword,
 			opts.Keyword)
 	}
 
